trl/extras/vllm_client.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2025 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import logging
+import time
+from typing import Optional
+
+import requests
+import torch
+from requests import ConnectionError
+from torch import nn
+from vllm.distributed.device_communicators.pynccl import PyNcclCommunicator
```

Comment:
Since `vllm` is not a core dep, do we need to have something like `is_vllm_available()` for these imports?